### PR TITLE
support multiple e

### LINKED ISSUE #2: Support multiple -e

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #2

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -12,21 +12,28 @@ import (
...
+func (*arrayFlags) String() string {
+	return ""
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+var flagTest arrayFlags
```
Comment (Al2Klimov) [inline main.go]: Why a **global** variable?

File: main.go [hunk 2]
```diff
@@ -12,21 +12,28 @@ import (
...
+
 func main() {
-	patternParameter := flag.String("e", "", "use PATTERN for matching.")
+	flag.Var(&flagTest, "e", "use several PATTERNS for matching.")
```
Comment (Al2Klimov) [inline main.go]: You don't need to change the description.

File: main.go [hunk 3]
```diff
@@ -40,11 +47,20 @@ func main() {
...
+				os.Exit(1)
+			}
+
+			if re.Match(data) {
+				if !bytes.HasSuffix(data, []byte{'\n'}) {
+					data = append(data, '\n')
+				}
+				_, _ = os.Stdout.Write(data)
+				break
 			}
```
Comment (Al2Klimov) [inline main.go]: You're compiling the regexes per **every single line**.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (justinmueller05082000) [reply main.go]: i could save my re's into an array before the big for loop but i dont know if its possible to use it like i do at the moment. for example if i got arr := []*Regexp{re, re1, re2} can i then do something like that? -> arr[0].Match(data)?

Comment (Al2Klimov) [reply main.go]: Try it out.

Comment (justinmueller05082000) [reply main.go]: if i cant i have absolutely no idea how to fix it :D

File: main.go [hunk 4]
```diff
@@ -40,13 +57,15 @@ func main() {
...
+			if i.Match(data) {
+				if !bytes.HasSuffix(data, []byte{'\n'}) {
+					data = append(data, '\n')
+				}
+				_, _ = os.Stdout.Write(data)
+				break
 			}
-			_, _ = os.Stdout.Write(data)
 		}
-
```
Comment (Al2Klimov) [inline main.go:49]: Not part of the task.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/support-multiple-e-2`

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toGrep.txt:**

```
foobar dog cat dog cat foobar zebra
elephant mouse dog cat fox
dragon fox elephant mouse dog cat
ant ant dog cat dog mouse
```

**Go code:**

**Terminal:** `go run . -e 'fo*bar' -e 'dragon' -e 'zebra' < toGrep.txt`

**Output:**

```
foobar dog cat dog cat foobar zebra
dragon fox elephant mouse dog cat
```

**Grep command:**

**Terminal:**  `grep -e 'fo*bar' -e 'dragon' -e 'zebra' < toGrep.txt`

**Output:**

```
foobar dog cat dog cat foobar zebra
dragon fox elephant mouse dog cat
```


**Conclusion:**
Both results are equal except for the colors.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**